SharpPcap/LibPcap/RemoteAuthentication.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Net;
+using System.Runtime.InteropServices;
+
+namespace SharpPcap.LibPcap
+{
+    /// <summary>
+    /// Remote authentication type and parameters
+    /// </summary>
+    public class RemoteAuthentication : ICredentials
```

Comment:
There is no need for yet another `RemoteAuthentication` class, simply use the `NetworkCredential` class

Reply:
@kayoub5 how does this map? The RemoteAuthentication class has 'Type' that maps through the remote authentication. If we use a NetworkCredential directly how do we map the 'Type'?

Reply:
It's controlled through [RemotePcap.CreateAuth](https://github.com/chmorgan/sharppcap/blob/master/SharpPcap/LibPcap/RemotePcap.cs#L11).

Examples:

Usage | Input | `AuthenticationTypes`
-------| ------|-----------------------
Implicit | `NetworkCredential()` | `Null`
Implicit | `NetworkCredential(null, null)` | `Null`
Implicit | `null` | `Null`
Explicit | `NetworkCredential("foo", "bar", "null")` | `Null`
Explicit | `NetworkCredential("foo", "bar", "0")` | `Null`
Implicit | `new NetworkCredential("user", "pass")` | `Password`
Explicit | `new NetworkCredential("user", "pass", "pwd")` | `Password`
Explicit | `new NetworkCredential("user", "pass", "1")` | `Password`

Reply:
Imo the implicit portion of the above table is going to make it more difficult for users to follow.

I'm also having trouble following why it makes sense to use ICredential or NetworkCredential. I don't see the benefit of using the ICredential when it doesn't map clearly to the remote auth stuff. If we used a RemoteAuthentication everywhere instead of an ICredential imo, and fold RemotePcap into this same class as a static method to keep the code closer together.

At that point you'd have something like an internal RemoteAuthentication.CreateAuth() with no parameters that would return a pcap_rmauth.

Thoughts?

Reply:
@chmorgan let's look at it this way
Input | `AuthenticationTypes`
------|-----------------------
`NetworkCredential(null, null)` | `Null`
`null` | `Null`
`new NetworkCredential("user", "pass")` | `Password`

This mapping will cover 99.99% of use cases.

Using builtin type `NetworkCredential` allows for better security, since password would be encrypted in memory.

Reply:
@kayoub5 well we do get the username and password back out for authentication, wouldn't that imply they are decrypted when we request them? I've got a PR almost ready that flattens things a bit that I *think* makes it easier to follow overall. The idea of using ICredentials and NetworkCredentials is a reasonable one but imo the concern is that we are effectively downcasting from an ICredentials back to a pcap_rmauth and the URI field maps oddly etc.

Reply:
@kayoub5 opened a PR with just these changes for review.

Reply:
@chmorgan the table I provided describes what's already supported by the code in master.